aeneas/src/ssa/SsaDebugger.v3

```diff
@@ -109,35 +139,48 @@ class SsaDebugger(prog: Program, interp: SsaInterpreter) {
 		}
 		return true;
 	}
+	// Start the program to be debugged in the interpreter
 	def start() {
 		interp.reset();
+		srcPrinter.reset();
 		Terminal.put1("Starting program: %s\n", interp.prog.name());
 		interp.pushFrame(startDel.memberRef, startDel.val, startArgs);
 		running = true;
 	}
-	def execute(untilBreak: bool, nextLine: bool, jump: bool) {
+	// Step debugged program and stop depending on condition
+	def execute(untilBreak: bool, nextLine: bool, jump: bool, fin: bool) {
 		if (!running) start(); // ???: start or return?
 		var targetSp = interp.sp;
+		var prev = interp.frame.ip;
 		var n = interp.step(interp.frame.ip);
 		if (interp.frame == null) {
 			running = false;
 			printExecuteResult();
 			return;
 		}
+		srcPrinter.putData(prev, targetSp);
 		interp.frame.ip = n;
-		while (!stopExecute(n, untilBreak, nextLine, jump, targetSp)) {
+		while (!stopExecute(n, untilBreak, nextLine, jump, fin, targetSp)) {
```

Comment:
The four booleans make this a little difficult to figure out what combinations are valid. Does an enum make sense for the stopping condition?